QHyper/converter.py

```diff
@@ -147,60 +148,32 @@ def to_dimod_qubo(problem: Problem, lagrange_multiplier: float = 10
         return cast(tuple[dict[tuple[str, ...], float], float],
                     bqm.to_qubo())  # (qubo, offset)
 
-    # TODO: Refactor is needed
-    @staticmethod
-    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
-        if hasattr(problem, "one_hot_encoding"):
-            if problem.one_hot_encoding:
-                return Converter._dqm_from_one_hot_representation(problem)
-            return Converter._dqm_from_discrete_representation(problem)
-        return Converter._dqm_from_one_hot_representation(problem)
 
     @staticmethod
-    def _dqm_from_discrete_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        for var in problem.variables:
-            var = str(var)
-            if var not in dqm.variables:
-                dqm.add_variable(problem.cases, var)
-
-        for vars, bias in problem.objective_function.as_dict().items():
-            x_i, x_j = vars
-            xi_idx: int = cast(int, dqm.variables.index(x_i))
-            xj_idx: int = cast(int, dqm.variables.index(x_j))
-
-            # We're skipping the linear terms
-            if xi_idx == xj_idx:
-                continue
-
-            dqm.set_quadratic(
-                dqm.variables[xi_idx],
-                dqm.variables[xj_idx],
-                {(case, case): bias for case in range(problem.cases)},
-            )
-
-        return dqm
-
-    @staticmethod
-    def _dqm_from_one_hot_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        CASES_OFFSET = 1 if problem.cases == 1 else 0
+    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
 
         def binary_to_discrete(v: str) -> str:
             id = int(v[1:])
             discrete_id = id // problem.cases
             return f"x{discrete_id}"
 
-        variables_discrete = [
-            binary_to_discrete(str(v)) for v in problem.variables[:: problem.cases]
-        ]
-        for var in variables_discrete:
-            if var not in dqm.variables:
-                dqm.add_variable(problem.cases + CASES_OFFSET, var)
+        dqm = dimod.DiscreteQuadraticModel()
+
+        objective_function_variables = sorted(problem.objective_function.get_variables(),
+                     key=lambda x: int(x[1:]))
```

Comment:
That will work only for variables like 'x1', 'y3', but not for 'xy1', and why exactly are you sorting these?
If you really need to change variables name then you will have to do some mapping I think.
And if you change them, then later they should be remapped to the original one, so user can understand which one is which

Reply:
Previously, `problem.variables` was used, where variables were stored in the form of, for example: `(s0, s1, s2, s3,...)`, sorted variables that were used in problem.
Due to the fact that this variable does not have to be implemented, I decided to extract the variables used from the objective function. From what I saw, each currently implemented problem has a `problem.variables` variable, so maybe we can return to this solution?

Reply:
>That will work only for variables like 'x1', 'y3', but not for 'xy1'

What about that?